### PR TITLE
feat: support non-error type "error.cause"

### LINKED ISSUE #934: [feat] Support non-error cause property
## What are the steps to reproduce this issue?

Throw an error with data inside `cause` property.

```
throw new Error('Name', { cause: { code: 'NonInteger' } })
```

## What happens?

Error `cause` is not visible on HoneyBadger web UI.

## What were you expecting to happen?

Error `cause` to be visible on Honeybadger web UI.

## Any other comments?

> The value can be of any type. You should not make assumptions that the error you caught has an Error as its cause

From [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause).

## What versions are you using?
**Operating System:** macOS
**Package Name:** @honeybadger-io/js
**Package Version:** 4.6.0
**Browser Version:** Chrome 107.0.0.0

### PR DESCRIPTION
**Pull request by subzero10**
## Status
**READY**

Closes: #934

## Description
Support non-error type `cause`.

### CODE REVIEW

File: packages/js/test/unit/server/middleware.server.test.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ describe('Express Middleware', function () {
       const initialContext = client.__getContext();
       client.setContext({ reqId: req.params.reqId });
       setTimeout(() => {
-        res.json({
+        res.status(200).json({
```
Comment (subzero10) [inline packages/js/test/unit/server/middleware.server.test.ts:103]: `res.json` is deprecated, it should be `res.status().json()`

**Comment by subzero10** [comment]
@joshuap I modified the js library to send non-error type `cause` to the backend, but it seems that the Honeybadger web interface expects to render an Error object.

Here's how it looks when the `cause` object has `class`, `message` and `backtrace` properties:
Screenshot 2022-11-12 at 5 56 31 PM

And here's how it looks when the `cause` object is missing those properties:
Screenshot 2022-11-12 at 6 06 01 PM
Screenshot 2022-11-12 at 6 05 53 PM

**Review by joshuap — CHANGES_REQUESTED** [review]
> @joshuap I modified the js library to send non-error type `cause` to the backend, but it seems that the Honeybadger web interface expects to render an Error object.
> 
> Here's how it looks when the `cause` object has `class`, `message` and `backtrace` properties: Screenshot 2022-11-12 at 5 56 31 PM
> 
> And here's how it looks when the `cause` object is missing those properties: Screenshot 2022-11-12 at 6 06 01 PM Screenshot 2022-11-12 at 6 05 53 PM

Edit: I'm still catching up on this, obviously. Let's chat about implementation on Slack—maybe we can figure out a way to also support the structured data use case. :)

**Comment by joshuap** [comment]
Sorry, there should have been extra clarification on #934 before starting on it.

**Comment by joshuap** [comment]
You could also ask if context would solve the use case in #934. If so, we could always skip this feature for now.

**Review by shalvah — APPROVED** [review]

**Comment by subzero10** [comment]
Closing because of inactivity. We may come back to this again.

**Pull request closed without merging**